Azure-Samples/modern-data-warehouse-dataops
# Bug Fix: Parking Sensor E2E - Bay_Id is not Unique in the Source Data, Causing Ingestion Pipeline to Create Dupes

# Type of PR
- Code changes

## Purpose
The PowerBI dashboard fails to load after the data ingestion job runs a second time, due to a duplicate identifier in the dim_parking_bay table.
![image](https://user-images.githubusercontent.com/49792089/199089132-46d4215c-dbc2-48a6-8d26-e5abb3ec4806.png)

This is caused by the Parking Sensor E2E code assuming that Bay_Id is a unique identifier for a parking bay sensor. However, the bay data contains entries that have the same Bay_Id (perhaps due to the changes noted on the [source website](https://data.melbourne.vic.gov.au/Transport/On-street-Parking-Bay-Sensors/vh2v-4nfs))? 
![image](https://user-images.githubusercontent.com/49792089/199101355-a83fe0b9-a745-4f56-bad9-ac545282d247.png)

Perhaps st_marker_id and bay_id comprise a compound key that's unique? This PR fixes the example so that it's functional (although potentially less accurate) by removing dupes in the bay_data. 

During the ingestion job, the transform stage does a left join on the bay + sensor data using the bay_id. This join will duplicate rows when 

## Author pre-publish checklist

- [X ] Added test to prove my fix is effective or new feature works

## Validation steps
1. Run ingest job multiple times
1. Refresh PowerBI data - without the fix, it will yield the "duplicate value" error above. With the fix, the refresh works.

----

Alternative:
Without the fix, observe that the metrics show the number of parking bays growing with each successive ingestion run:
![image](https://user-images.githubusercontent.com/49792089/199124860-8658a522-12e7-45ac-bef8-45832d76323a.png)

However, it remains constant (as expected) across runs after the fix is applied.

Kusto Query:

```bash
customEvents
| order by timestamp desc
| where name startswith "Transform (ai): Completed"
| project timestamp, appName, name,
    pipelineName             = customDimensions.pipeline,
    pipelineRunId            = customDimensions.run_id,
    dimParkingbayCount       = customDimensions.new_parkingbay_count

```
